### PR TITLE
API: Add `asarray` function

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @hameerabbasi,

This PR adds [`asarray` Array API function](https://data-apis.org/array-api/latest/API_specification/generated/array_api.asarray.html).

### CODE REVIEW

File: src/finch/tensor.py [hunk 1]
```diff
@@ -412,11 +416,36 @@ def random(shape, density=0.01, random_state=None):
...
 
+def asarray(obj, /, *, dtype=None, format=None):
+    if format not in {"coo", "csr", "csc", "csf", "dense", None}:
+        raise ValueError(f"{format} format not supported.")
+    tensor = obj if isinstance(obj, Tensor) else Tensor(obj)
+
+    if format is not None:
+        order = tensor.get_order()
+        if format == "coo":
+            storage = Storage(SparseCOO(tensor.ndim, Element(tensor.fill_value)), order)
```
Comment (hameerabbasi) [inline src/finch/tensor.py:427]: Will `Tensor.ndim` work for this one?

File: src/finch/tensor.py [hunk 2]
```diff
@@ -412,11 +416,36 @@ def random(shape, density=0.01, random_state=None):
...
+        elif format == "csr":
+            storage = Storage(Dense(SparseList(Element(tensor.fill_value))), order)
+        elif format == "csc":
+            storage = Storage(Dense(SparseList(Element(tensor.fill_value))), order)
+        elif format == "csf":
+            storage = Storage(Dense(SparseList(SparseList(Element(tensor.fill_value)))), order)
```
Comment (hameerabbasi) [inline src/finch/tensor.py]: CSF is NDim-agnostic, it's just `Dense` followed by however many `SparseList` we need.

**Review by hameerabbasi — COMMENTED** [review]
A couple of questions.

File: src/finch/tensor.py (hunk 1 above)
Comment (mtsokol) [reply src/finch/tensor.py:427]: Here we need to provide dimensionality of the input tensor. `ndim` is an instance attribute, not class attribute, so `Tensor.ndim` won't work here.

File: src/finch/tensor.py (hunk 2 above)
Comment (mtsokol) [reply src/finch/tensor.py]: Right - updated!

File: src/finch/tensor.py (hunk 1 above)
Comment (hameerabbasi) [reply src/finch/tensor.py:427]: I mean that would NDim work for a COO-format tensor?

Comment (hameerabbasi) [reply src/finch/tensor.py:427]: Resolved in Slack; `ndim` would do the right thing.

**Review by hameerabbasi — APPROVED** [review]

**Pull request merged**